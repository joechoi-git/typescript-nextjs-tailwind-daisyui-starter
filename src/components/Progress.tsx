import React from "react";

interface ProgressProps {
    value: number;
}

export default function Progress({ value }: ProgressProps): React.JSX.Element {
    return (
        <div
            className="radial-progress bg-primary text-primary-content border-4 border-primary"
            style={{ "--value": value } as React.CSSProperties}
            role="progressbar"
        >
            {value}%
        </div>
    );
}
