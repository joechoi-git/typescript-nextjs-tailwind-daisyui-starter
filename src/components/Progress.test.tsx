import React from "react";
import { render } from "@testing-library/react";
import Progress from "./Progress";

describe("Progress Component", () => {
    it("should render with the correct value", () => {
        const value = 50;
        const { getByRole } = render(<Progress value={value} />);
        const progressBar = getByRole("progressbar");

        expect(progressBar).toBeInTheDocument();
        expect(progressBar).toHaveStyle(`--value: ${value}`);
        expect(progressBar).toHaveTextContent(`${value}%`);
    });

    it("should render with the correct value at 0", () => {
        const value = 0;
        const { getByRole } = render(<Progress value={value} />);
        const progressBar = getByRole("progressbar");

        expect(progressBar).toBeInTheDocument();
        expect(progressBar).toHaveStyle(`--value: ${value}`);
        expect(progressBar).toHaveTextContent(`${value}%`);
    });

    it("should render with the correct value at 100", () => {
        const value = 100;
        const { getByRole } = render(<Progress value={value} />);
        const progressBar = getByRole("progressbar");

        expect(progressBar).toBeInTheDocument();
        expect(progressBar).toHaveStyle(`--value: ${value}`);
        expect(progressBar).toHaveTextContent(`${value}%`);
    });

    it("should match snapshot", () => {
        const value = 50;
        const { asFragment } = render(<Progress value={value} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
