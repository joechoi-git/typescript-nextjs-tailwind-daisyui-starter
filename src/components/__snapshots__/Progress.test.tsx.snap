// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="radial-progress bg-primary text-primary-content border-4 border-primary"
    role="progressbar"
    style="--value: 50;"
  >
    50%
  </div>
</DocumentFragment>
`;
